app/scripts/modules/amazon/src/securityGroup/configure/configSecurityGroup.mixin.controller.js

```diff
@@ -337,6 +337,15 @@ module(AMAZON_SECURITYGROUP_CONFIGURE_CONFIGSECURITYGROUP_MIXIN_CONTROLLER, [
       ruleset.splice(index, 1);
     };
 
+    ctrl.updateRuleType = function(type, ruleset, index) {
+      const rule = ruleset[index];
+      rule.type = type;
```

Comment:
I don't remember enough of how Angular works - is this assignment needed because you're implementing an `ng-change` handler?

Reply:
Yes exactly. Otherwise, we are not intercepting this change as it saves and cannot dynamically render the port number

Reply:
It looks unnecessary to me as well because of the `ng-model` binding

Reply:
When a new rule is added, it defaults to TCP type rule with port 7001 as the default. So `ng-model` is referencing those value. There is no event listener to automatically update that model when rule type changes.

Another alternative is to render 2 inputs, but I thought adding a method would be cleaner if we need to extend the rules later.

```
<input ng-if="rule.type === 'icmp'" />
<input ng-if="rule.type !== 'icmp'" />
```

Reply:
Just to confirm (the Angular stuff is always confusing): it looks like this is what's happening:
* the `ng-change` handler fires before the value is set on the `ng-model`, which is why you need to pass the new value in to the handler
* after the `ng-change` handler completes, the `ng-model` sets the value to the new value

Does that sound right? If so, this looks great!

Reply:
Pretty much!
- `ng-change` fires when the value is updated. The target value has already been set (`ruleType` in this case), but none of the other values changed until `ng-chagne` runs. 
- after the handler completes, `ng-model` sets the fields to the new values